src/irmin-layers/irmin_layers_intf.ml

```diff
@@ -20,31 +20,30 @@ module type S = sig
   include Irmin.S
 
   val freeze :
-    ?min:commit list ->
-    ?max:commit list ->
-    ?squash:bool ->
-    ?copy_in_upper:bool ->
+    ?min_lower:commit list ->
+    ?max_lower:commit list ->
     ?min_upper:commit list ->
+    ?max_upper:commit list ->
     ?recovery:bool ->
     repo ->
     unit Lwt.t
-  (** [freeze ?min ?max ?squash ?copy_in_upper ?min_upper ?recovery t] launches
+  (** [freeze ?min_lower ?max_lower ?min_upper ?max_upper ?recovery t] launches
       an asynchronous freezing operation on the repo [t] to reduce the size of
       the upper layer and discard unnecessary branches of objects (i.e. commits,
       nodes and contents).
 
-      Let [o] be the set of objects reachable from the [max] commits and bounded
-      by the [min] commits. During the freeze, all objects in [o] are copied to
-      the lower layer, if there is one. Setting [squash] to [true] is equivalent
-      to setting [min] to [max]. [min] defaults to the empty list and [max]
-      defaults to the head commits of the repo.
+      Let [o] be the set of objects reachable from the [max_lower] commits and
+      bounded by the [min_lower] commits. During the freeze, all objects in [o]
+      are copied to the lower layer, if there is one. [min_lower] defaults to
+      the empty list (i.e. the copy is unbounded) and [max_lower] defaults to
+      the head commits of the repo. When [max_lower] is the empty list, nothing
+      is copied.
 
-      Let [o'] be the set of objects reachable from the [max] commits and
-      bounded by the [min_upper] commits. When the freeze is over, if
-      [copy_in_upper] is true, then the new upper layer will only contain the
-      objects of [o'], otherwise the new upper layer will be empty. [min_upper]
-      defaults to the empty list and [copy_in_upper] defaults to the repo's
-      configuration.
+      Let [o'] be the set of objects reachable from the [max_upper] commits and
+      bounded by the [min_upper] commits. When the freeze is over, the new upper
+      layer will only contain the objects of [o']. [min_upper] defaults to the
+      empty list and [max_upper] defaults to [max_lower]. When [max_upper] is
```

Comment:
maybe it's worth to be more specific here: when the `min_upper` is empty, we only copy the `max_upper` (https://github.com/mirage/irmin/blob/master/src/irmin-pack/layered/ext_layered.ml#L650) and so its not the same behaviour as for lower.

Reply:
I completely forgot about that line!

Let's do the following:
- If the user doesn't provide a `min_upper`, set `min_upper` to `max_upper` (same as now)
- If the user provides a non-empty `min_upper`, use it (same as now)
- If the user provides an empty `min_upper`, use it (different from now)

What do you think?

Reply:
Nice, this way `min_upper` and `min_lower` have the same semantics (just different defaults).